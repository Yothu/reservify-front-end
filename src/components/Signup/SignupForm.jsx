import { useEffect, useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { useForm } from 'react-hook-form';
import { Oval } from 'react-loader-spinner';
import { toast } from 'react-toastify';
import Button from '../Ui/Button';
import ErrorMsg from '../Ui/ErrorMsg';
import { userRegister, reset } from '../../redux/auth/auth-slice';

const SignupForm = () => {
  const {
    register,
    handleSubmit,
    watch,
    formState: { errors },
  } = useForm();

  const password = useRef({});
  password.current = watch('password');

  const navigate = useNavigate();
  const dispatch = useDispatch();

  // prettier-ignore
  const {
    user,
    isLoading,
    isSuccess,
    isError,
    message,
  } = useSelector((state) => state.auth);

  useEffect(() => {
    if (isError) {
      toast.error(message);
    }
    if (isSuccess || user) {
      toast.success('Successfully registered');
      navigate('/login');
    }
    dispatch(reset());
  }, [user, isError, isSuccess, message, navigate, dispatch]);

  const onSubmit = (data) => {
    dispatch(userRegister(data));
  };

  if (isLoading) {
    return (
      <div className="d-flex justify-content-center">
        <Oval color="#FBBC05" height={250} width={250} />
      </div>
    );
  }

  return (
    <form
      className="d-flex flex-column align-items-center gap-2 container"
      onSubmit={handleSubmit(onSubmit)}
    >
      <input
        type="text"
        placeholder="Username"
        className="form-control"
        {...register('username', { required: 'Username is required' })}
      />
      {errors.username && <ErrorMsg message={errors.username.message} />}
      <input
        type="text"
        placeholder="Email"
        className="form-control"
        {...register('email', {
          required: 'Email is required',
          pattern: {
            value: /\S+@\S+\.\S+/,
            message: 'Invalid email address',
          },
        })}
      />
      {errors.email && <ErrorMsg message={errors.email.message} />}
      <input
        type="password"
        placeholder="Password"
        className="form-control"
        {...register('password', {
          required: 'Password is Required',
          minLength: { value: 6, message: 'Password is Too Short' },
        })}
      />
      {errors.password && <ErrorMsg message={errors.password.message} />}
      <input
        type="password"
        placeholder="Confim Password"
        className="form-control"
        {...register('confirm_password', {
          validate: (value) => {
            if (value === password.current) return null;
            return 'The passwords do not match';
          },
        })}
      />
      {errors.confirm_password && (
        <ErrorMsg message={errors.confirm_password.message} />
      )}
      <Button type="submit" text="Signup" cName="mt-3" />
    </form>
  );
};

export default SignupForm;
