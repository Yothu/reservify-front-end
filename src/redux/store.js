import { configureStore } from '@reduxjs/toolkit';
import hotelReducer from './hotels/hotel-slice';
import authReducer from './auth/auth-slice';
import reservationReducer from './reservations/reservation-slice';
import locationReducer from './locations/location-slice';

const store = configureStore({
  reducer: {
    auth: authReducer,
    hotel: hotelReducer,
    reservation: reservationReducer,
    location: locationReducer,
  },
});

export default store;
