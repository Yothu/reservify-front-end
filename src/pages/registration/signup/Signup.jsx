import SignupForm from '../../../components/Signup/SignupForm';
import registerImg from '../../../assets/images/undraw-svg/green-register.svg';
import style from './Signup.module.css';

function Signup() {
  return (
    <section className={`${style.signup} conatiner`}>
      <div className="text-center mb-3">
        <img src={registerImg} alt="register" className="mw-100" width={500} />
      </div>

      <SignupForm />
    </section>
  );
}

export default Signup;
