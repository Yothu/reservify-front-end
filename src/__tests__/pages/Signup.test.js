import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import Signup from '../../pages/registration/signup/Signup';

describe('Login Page', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Signup />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
