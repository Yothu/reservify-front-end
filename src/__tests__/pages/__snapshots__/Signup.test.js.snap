// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="signup conatiner"
      >
        <div
          class="text-center mb-3"
        >
          <img
            alt="register"
            class="mw-100"
            src="green-register.svg"
            width="500"
          />
        </div>
        <form
          class="d-flex flex-column align-items-center gap-2 container"
        >
          <input
            class="form-control"
            name="username"
            placeholder="Username"
            type="text"
          />
          <input
            class="form-control"
            name="email"
            placeholder="Email"
            type="text"
          />
          <input
            class="form-control"
            name="password"
            placeholder="Password"
            type="password"
          />
          <input
            class="form-control"
            name="confirm_password"
            placeholder="Confim Password"
            type="password"
          />
          <button
            class="button mt-3"
            type="submit"
          >
            Signup
          </button>
        </form>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="signup conatiner"
    >
      <div
        class="text-center mb-3"
      >
        <img
          alt="register"
          class="mw-100"
          src="green-register.svg"
          width="500"
        />
      </div>
      <form
        class="d-flex flex-column align-items-center gap-2 container"
      >
        <input
          class="form-control"
          name="username"
          placeholder="Username"
          type="text"
        />
        <input
          class="form-control"
          name="email"
          placeholder="Email"
          type="text"
        />
        <input
          class="form-control"
          name="password"
          placeholder="Password"
          type="password"
        />
        <input
          class="form-control"
          name="confirm_password"
          placeholder="Confim Password"
          type="password"
        />
        <button
          class="button mt-3"
          type="submit"
        >
          Signup
        </button>
      </form>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
